File: 819#MostCommonWord.py
# 819. Most Common Word
class Solution:
    def mostCommonWord(self, paragraph, banned):
        """
        :type paragraph: str
        :type banned: List[str]
        :rtype: str
        """
        paragraph = ''.join(map(str.lower, paragraph))
        for p in "!?',;.":
            paragraph = paragraph.replace(p, "")
        ls = paragraph.split(" ")
        words = set(ls)
        res = None
        count = 0
        for word in words:
            if word not in banned:
                temp = ls.count(word)
                if temp > count:
                    res = word
                    count = temp
        return res

if __name__ == "__main__":

    para = "Given a Words in the list of banned words are given in lowercase, and free of punctuation.  Words in the paragraph are not case sensitive. The answer is in lowercase."

    so = Solution()
    banned = ["of", "is", "are"]
    res = so.mostCommonWord(para, banned)
    print(res)

File: 819#MostCommonWord.go
// 819. Most Common Word
package main

import "strings"

func uniqueList(slist []string) []string {
	unique := make([]string, 0)
	for _, w := range slist {
		if isInside(unique, w) {
			continue
		} else {
			unique = append(unique, w)
		}
	}
	return unique
}

func isInside(slice []string, word string) bool {
	for _, w := range slice {
		if w == word {
			return true
		}
	}
	return false
}

// Count helper function
func Count(slice []string, target string) int {
	count := 0
	for _, word := range slice {
		if word == target {
			count++
		}
	}
	return count
}

func mostCommonWord(paragraph string, banned []string) string {
	paragraph = strings.ToLower(paragraph)
	for _, p := range []string{"!", "?", "'", ";", ".", ","} {
		paragraph = strings.Replace(paragraph, p, "", -1)
	}
	ls := strings.Split(paragraph, " ")
	words := uniqueList(ls)
	var res string
	count := 0
	for _, word := range words {
		if !isInside(banned, word) {
			temp := Count(ls, word)
			if temp > count {
				res = word
				count = temp
			}
		}
	}
	return res

}
